psh/tools/common.py

```diff
@@ -38,6 +38,25 @@ def create_testdir(p, dirname):
     psh.assert_cmd(p, f'mkdir {dirname}', '', msg)
 
 
+def assert_timestamp_change(p, touched_files, uptimes, dir=''):
+    files = psh.ls(p, dir)
+    for file in files:
+        if file.name in touched_files:
+            uptime = uptimes[file.name]
```

Comment:
not checking if all files from `touched_files` are present in `ls`. IMHO the loop should be over `touched_files` not the other way around